cmd/output/task_test.go

```diff
@@ -0,0 +1,178 @@
+package output
+
+import (
+	"bytes"
+	"errors"
+	"io"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var errTest = errors.New("test error")
+
+var _ terminal = &stubTerminal{}
+
+type stubTerminal struct {
+	width         int
+	isNotTerminal bool
+	buf           *bytes.Buffer
+	getSizeErr    error
+}
+
+func (s *stubTerminal) Writer() io.Writer {
+	return s.buf
+}
+
+func (s *stubTerminal) GetSize() (int, int, error) {
+	return s.width, 0, s.getSizeErr
+}
+
+func (s *stubTerminal) IsTerminal() bool {
+	return !s.isNotTerminal
+}
+
+func TestTask(t *testing.T) {
+	type testCase struct {
+		msg           string
+		termWidth     int
+		expected      string
+		getSizeErr    error
+		isNotTerminal bool
+	}
+
+	newStubTerminal := func(tc testCase, buf *bytes.Buffer) *stubTerminal {
+		return &stubTerminal{isNotTerminal: tc.isNotTerminal, width: tc.termWidth, buf: buf, getSizeErr: tc.getSizeErr}
+	}
+
+	testCases := []testCase{
+		{
+			msg:       "message that is too long is trimmed",
+			termWidth: 20,
+			expected:  " message th" + AnsiFaint + "..." + AnsiReset,
+		},
+
+		{
+			msg:       "a message that is not too long is not trimmed",
+			termWidth: 60,
+			expected:  " a message that is not too long is not trimmed" + AnsiFaint + "........" + AnsiReset,
+		},
+		{
+			msg:       "even without message the line is too narrow cannot be trimmed",
+			termWidth: 1,
+			expected:  " " + AnsiFaint + "..." + AnsiReset,
+		},
+		{
+			msg:       "no matter the terminal width the maximum width is not exceeded",
+			termWidth: 1000,
+			expected:  " no matter the terminal width the maximum width is not exceeded" + AnsiFaint + "..................................................." + AnsiReset,
+		},
+		{
+			msg:        "uses fallback width with GetSize error",
+			expected:   " uses fallback width with GetSize error" + AnsiFaint + "..............." + AnsiReset,
+			getSizeErr: errTest,
+		},
+		{
+			msg:           "uses fallback width when not terminal error",
+			expected:      " uses fallback width when not terminal error" + AnsiFaint + ".........." + AnsiReset,
+			isNotTerminal: true,
+		},
+	}
+	t.Run("StartTaskWithTerminal", func(t *testing.T) {
+		t.Run("start task writes start line with hourglass", func(t *testing.T) {
+			for _, tc := range testCases {
+				t.Run(tc.msg, func(t *testing.T) {
+					buf := bytes.NewBuffer(nil)
+					term := newStubTerminal(tc, buf)
+
+					StartTaskWithTerminal(tc.msg, term)
+
+					actual := buf.String()
+					expected := hourglass + tc.expected
```

Comment:
I was just a little confused at to where hourglass came from. Could we consider making a package called icons (could be a subpackage of output), that could contain these, so it would be used `icons.hourglass`? Maybe it's out of scope for this specific PR, just something to have in mind 😄

Reply:
I guess we could rename it `iconHourglass`, but yeah it is not something that is changed in this PR